flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/optimization/Regularization.scala

```diff
@@ -95,6 +95,7 @@ abstract class DiffRegularization extends Regularization {
       regParameter: Double)
 }
 
+// TODO(tvas): I think NoRegularization should extend DiffRegularization
```

Comment:
Because the idea is to disallow setting non-differentiable regularization in solver like L-BFGS which require differentiable loss, through type-matching.

By setting NoRegularization to extend DiffRegularization we would be able to use it with L-BFGS